modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,46 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
-
 				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+
+					if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
+						continue;
+					}
+
+					var sizes = media[i].media_details.sizes;
+					var sizes_keys = Object.keys( sizes );
 					var srcset = images[j].getAttribute( 'srcset' );
-					if ( srcset ) {
-						images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+
+					// If a full image is present in srcset, it should be updated.
+					if( srcset && media[i].media_details.sources['image/webp'] ) {
+						srcset = srcset.replace( media[i].media_details.sources['image/webp'].file, media[i].media_details.sources['image/jpeg'].file );
+					}
+
+					var flag = true;
+					for ( var k = 0; k < sizes_keys.length; k++ ) {
+						if( ! media[i].media_details.sizes[sizes_keys[k]].sources || ! media[i].media_details.sizes[sizes_keys[k]].sources['image/webp'] || ! media[i].media_details.sizes[sizes_keys[k]].sources['image/jpeg'] ) {
```

Comment:
media[i].media_details.sizes[sizes_keys[k]].sources is used repeatedly in this block, I would set a variable to this value at the top and use that instead

Reply:
It makes sense. Set variables for the main sources `media_sources = media[ i ].media_details.sources;` as well as size sources `media_sizes_sources = media[ i ].media_details.sizes[ sizes_keys[ k ] ].sources;`